More basic usages

chrismiceli opened the pull request:
The main goal here is to remove jQuery for more usages that don't involve creating and modifying a bunch of DOM elements. This mostly consist of
1. Using normal event listeners
2. Removing $.Callbacks in favor of arrays of methods
3. Removing $.isPlainObject

I also took the opportunity to fix a deprecation warning on event.which.

Again, this is low priority with some risks, so no pressure to merge or review.

Branch for testing
https://www.pgdp.org/~chrismiceli/c.branch/jquery/

70ray commented on scripts/splitControl.js:
```diff
@@ -58,8 +60,8 @@ var splitControl = function(container, config) {
 
-    let reSize = $.Callbacks();
-    let dragEnd = $.Callbacks();
+    let reSize = [];
```
Alternatively, here and elsewhere, you could replace $.Callbacks with a Set instead of an array. Then the associated add() functions could remain unchanged instead of changing to push(). (It is avaialable within our range of supported browsers.)

70ray replied:
Perhaps also rename reSize() to onResize() (and prepend on- similarly elsewhwere) to make it clearer what this is doing.

70ray commented on tools/proofers/previewControl.js:
```diff
@@ -179,16 +179,23 @@ $( function() {
-                dataType: "script",
-                cache: true,
-                url: "https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js",
+            const maxjaxScriptElement = document.createElement('script');
```
perhaps "mathJaxScriptElement" ?

70ray reviewed (commented):
By chance, I was also looking at some changes in the splitControl area. The only one which coincides with this is replacing the jquery callbacks which I have commented on.

I have tested tahat the math preview works.

cpeel reviewed (approved):
I'm a JS n00b and am fine with these changes as long as Ray is and it passes Sharon's testing.

70ray reviewed (approved):
Looks good to me.

srjfoo reviewed (changes requested):
Not entirely sure where this comment belongs in the code, so I'm not attaching it to any specific piece of code. In testing the splitter (and it's not just the splitter, it's text-only mode, as well), if I set the font size to other than browser default and refresh the page, the new font size persists. However, if I set a different font face for the text and refresh the page, both the font face and the font size are reset to browser default.

In going to test it in vertical mode (side-by-side) as well as horizontal mode (top-and-bottom), I discovered that when I refreshed the page in vertical mode, the display switched back to horizontal mode with font and font size back at browser default (note that when I switched back to vertical mode, the font size had persisted -- **_unless_** I also changed the font to other than browser default.

70ray commented:
> Not entirely sure where this comment belongs in the code, so I'm not attaching it to any specific piece of code. In testing the splitter (and it's not just the splitter, it's text-only mode, as well), if I set the font size to other than browser default and refresh the page, the new font size persists. However, if I set a different font face for the text and refresh the page, both the font face and the font size are reset to browser default.
> 
> In going to test it in vertical mode (side-by-side) as well as horizontal mode (top-and-bottom), I discovered that when I refreshed the page in vertical mode, the display switched back to horizontal mode with font and font size back at browser default (note that when I switched back to vertical mode, the font size had persisted -- **_unless_** I also changed the font to other than browser default.

It looks like (at least part of) the problem is with numbers being stored as strings. In the case of the split percent this is because the function uses toFixed() which gives a string.
Sorry I didn't notice this before. There must have been something sub-conscious which didn't surface because when I saw this comment I went straight to look at this.

chrismiceli commented:
> > Not entirely sure where this comment belongs in the code, so I'm not attaching it to any specific piece of code. In testing the splitter (and it's not just the splitter, it's text-only mode, as well), if I set the font size to other than browser default and refresh the page, the new font size persists. However, if I set a different font face for the text and refresh the page, both the font face and the font size are reset to browser default.
> > In going to test it in vertical mode (side-by-side) as well as horizontal mode (top-and-bottom), I discovered that when I refreshed the page in vertical mode, the display switched back to horizontal mode with font and font size back at browser default (note that when I switched back to vertical mode, the font size had persisted -- **_unless_** I also changed the font to other than browser default.
> 
> It looks like (at least part of) the problem is with numbers being stored as strings. In the case of the split percent this is because the function uses toFixed() which gives a string. Sorry I didn't notice this before. There must have been something sub-conscious which didn't surface because when I saw this comment I went straight to look at this.

Yeah, I saw this in the settings for the split percents too. I'll take a look and ensure there aren't any other instances that can be impacted. It is introduced when I enhanced the strength of the test on the values coming from storage. Before we just checked they were objects, I checked the properties were of the right type.

chrismiceli commented:
> > > Not entirely sure where this comment belongs in the code, so I'm not attaching it to any specific piece of code. In testing the splitter (and it's not just the splitter, it's text-only mode, as well), if I set the font size to other than browser default and refresh the page, the new font size persists. However, if I set a different font face for the text and refresh the page, both the font face and the font size are reset to browser default.
> > > In going to test it in vertical mode (side-by-side) as well as horizontal mode (top-and-bottom), I discovered that when I refreshed the page in vertical mode, the display switched back to horizontal mode with font and font size back at browser default (note that when I switched back to vertical mode, the font size had persisted -- **_unless_** I also changed the font to other than browser default.
> > 
> > 
> > It looks like (at least part of) the problem is with numbers being stored as strings. In the case of the split percent this is because the function uses toFixed() which gives a string. Sorry I didn't notice this before. There must have been something sub-conscious which didn't surface because when I saw this comment I went straight to look at this.
> 
> Yeah, I saw this in the settings for the split percents too. I'll take a look and ensure there aren't any other instances that can be impacted. It is introduced when I enhanced the strength of the test on the values coming from storage. Before we just checked they were objects, I checked the properties were of the right type.

Fixed the one instance where this could happen. The other places look good to me, but let me know if I missed somewhere.

70ray commented on scripts/page_browse.js:
```diff
@@ -145,7 +151,7 @@ function makeTextWidget(container, splitter = false, reLayout = null) {
     const storageKeyLayout = storageKey + "-layout";
     let layout = JSON.parse(localStorage.getItem(storageKeyLayout));
-    if(!$.isPlainObject(layout)) {
+    if(!layout || layout.splitDirection !== "horizontal" || layout.splitDirection !== "vertical") {
```
Should be
`if(!layout || (layout.splitDirection !== "horizontal" && layout.splitDirection !== "vertical")) {`
as you have elsewhere.

70ray reviewed (commented):
What you have is very rigorous but I think it would be sufficient to test the JSON.parse and set defaults like you have in show_word_context.js.

srjfoo commented:
It looks like the font and font size are persistent now, but if I switch from horizontal mode to vertical mode, and refresh the page, it's still switching back to horizontal mode when displaying image+text. This does not happen on the main TEST site.

70ray commented:
> It looks like the font and font size are persistent now, but if I switch from horizontal mode to vertical mode, and refresh the page, it's still switching back to horizontal mode when displaying image+text. This does not happen on the main TEST site.

The change I suggested above should have fixed that. Looks like Chris forgot to update the sandbox.

chrismiceli commented:
> > It looks like the font and font size are persistent now, but if I switch from horizontal mode to vertical mode, and refresh the page, it's still switching back to horizontal mode when displaying image+text. This does not happen on the main TEST site.
> 
> The change I suggested above should have fixed that. Looks like Chris forgot to update the sandbox.

updated now, busy weekend :).
